doc/design/session.md

```diff
@@ -0,0 +1,62 @@
+# Design Doc: Session
+
+## Abstract
+
+This design doc proposes to have an object called *Session* which
+encapsulates the environment in which the computation graph is
+executed.
+
+## Background
+
+A computation graph is executed in an environment which contains the
+[scope](./scope.md) and other states. PaddlePaddle used to only have
+an implicit global session on which `paddle.eval()` is executed.
+
+This has the limitation that the user can not create two independent
```

Comment:
Here is a broken logic.  It claimed but didn't explain why users cannot have two environments. The second sentence is to claim that it is necessary to have two environments.

From the text above, readers would ask it seems that what we need is two Scope instances, but not a new class Session.

Reply:
Thanks! Explained why user cannot have two environments, and also changed wording so that reader will know we need a new class Session.